lib/data/database_controller.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:flutter/widgets.dart';
+import 'package:path/path.dart';
+import 'package:sqflite/sqflite.dart';
+import 'package:to_do_flutter/data/shopping_list_data.dart';
+
+class DatabaseController {
+  Future<Database> database() async {
+    WidgetsFlutterBinding.ensureInitialized();
+
+    return openDatabase(
+      join(await getDatabasesPath(), 'shoppingApp.db'),
+      onCreate: (db, version) async {
+        await db.execute(
+            'CREATE TABLE shoppingList(id INTEGER PRIMARY KEY, title TEXT)');
+        await db.execute(
+            'CREATE TABLE productList(id INTEGER PRIMARY KEY, listId INTEGER, isChecked BOOL, name TEXT)');
+
+        return;
+      },
+      version: 1,
+    );
+  }
+
+  Future<int> insertList(ShoppingList list) async {
+    int listId = 0;
+    Database db = await database();
+    listId = await db
+        .insert('shoppingList', list.toMap(),
+            conflictAlgorithm: ConflictAlgorithm.replace)
+        .then((value) => listId = value);
+    return listId;
+  }
+
+  Future<void> updateListTitle(int id, String title) async {
+    Database db = await database();
+    await db
+        .rawUpdate("UPDATE shoppingList SET title = '$title' WHERE id = '$id'");
+  }
```

Comment:
na przykładzie tej funkcji pokażę jak można skrócić ten zapis:
```suggestion
Future<void> updateListTitle(int id, String title) async => database
      .rawUpdate("UPDATE shoppingList SET title = '$title' WHERE id = '$id'");
```
w tym przypadku nawet nie trzeba pisać await, jest to fajny skrócony zapis